src/security/KeyData.cc

```diff
@@ -78,13 +79,49 @@ Security::KeyData::loadX509CertFromFile()
     return bool(cert);
 }
 
+/**
+ * If possible, add a CA certificate as next in the X.509 chain of trust.
+ * Otherwise, report but otherwise ignore the failure and add nothing.
+ */
+void
+Security::KeyData::tryAddChainCa(const Security::CertPointer &ca)
+{
+#if TLS_CHAIN_NO_SELFSIGNED
+    // self-signed certificates are not valid in a sent chain
+    if (CertIsSelfSigned(ca)) {
+        debugs(83, DBG_PARSE_NOTE(2), "CA " << CertSubjectName(ca) << " is self-signed, will not be chained.");
+        return;
+    }
+#endif
+
+    CertPointer latestCert(chain.size() > 0 ? chain.front() : cert);
+
+    // checks that the chained certs are actually part of a chain for validating cert
```

Comment:
The old comment seems a bit misleading/awkward in the new code, while the code is self-documenting. Please consider removing this comment.

```suggestion
```